runtime/dynstats.c

```diff
@@ -621,6 +866,11 @@ dynstats_inc(dynstats_bucket_t *b, uchar* metric) {
 			STATSCOUNTER_INC(b->ctrOpsOverflow, b->mutCtrOpsOverflow);
 		}
 	}
+
+	if(b->persistStateInterval > 0 && ++b->nUpdates >= b->persistStateInterval) {
+		persistBucketState(b);
```

Comment:
I'd do this in a bg-thread. This can potentially cause unwanted latency spikes in log-event processing.

Reply:
ok